primer/brand
# Add escape key handlers to open accordions and its inner content

## Summary

Resolves https://github.com/primer/brand/issues/519


## List of notable changes:



- **adds** `esc` key press handler to open accordions titles, which both closes them and returns focus
- **adds** `esc` key press handler to accordion content elements, which both closes the accordion and returns focus to the nearest toggle ancestor. 


## What should reviewers focus on?



- Verify that the escape key event works as expected to close the accordions, whether it's on the toggle itself or inner link.

## Steps to test:



1. Open this story
1. Use keyboard navigation to focus on a accordion summary
1. Open accordion details using space or enter keys
1. Press escape to close the accordion details 
1. Optionally test inner link contained in details, which should similarly close the accordion and return focus

## Supporting resources (related issues, external links, etc):

- https://github.com/github/accessibility-audits/issues/7115

## Contributor checklist:


## Reviewer checklist:


## Screenshots:

Escape key press event on accordion summaries:

https://github.com/primer/brand/assets/13340707/7fed3c86-aed8-4f2e-8d07-d2f50cdffd5b



Escape key press events on internal links on open accordions:

https://github.com/primer/brand/assets/13340707/52336d57-d882-4f73-b6a5-5f5eaedea294
